### DISCUSSION TITLE
Persistent scratch buffer

### DISCUSSION THREAD

**Discussion by Daeraxa** [zed-industries/zed#24231] [Feature Requests]
### Check for existing issues

- [X] Completed

### Is your feature request related to a problem?

Would be nice to have an auto-saving scratch buffer that persists across windows and sessions.
Zed starts with your standard 'untitled' buffer but this is only relevant to the current window and is lost on close unless manually saved.
The reasoning behind this is to have somewhere to jot down notes or keep snippets of data for re-use (not full snippet support, just a generic temporary storage).

### Describe the solution you'd like

On a new window (or new pane) the scratch buffer is opened (as an optional alternative to the standard `untitled` buffer). Could also be manually opened in any existing pane. 

The buffer would save automatically and would persist over sessions and windows (i.e. if I have two Zed windows then the scratch is shared between them). It would remain 'pinned' to the top of the project list (if desired).

Not sure how Zed would detect the language used unless a setting is added to manually mark the language of current buffer.
If I wanted to be really fancy then it should be able to detect URLs and make them clickable.

### If applicable, add mockups / screenshots to help present your vision of the feature

![image](https://user-images.githubusercontent.com/58074586/180230368-6653c369-79fe-44c7-8a3e-19fcf2ec2ff9.png)

**Comment by bersace** [comment]
Doom Emacs binds scratch buffer to `space x`.

I usually have an unversionned markdown file in each project. Like a personnal README. Having it per project is very useful before triggering an issue for every idea in each project.

**Comment by probably-neb** [comment]
This is a neat idea and something worth keeping around as an idea/feature. We've decided to move feature requests to discussions so I'm going to move this there

**Comment by nkapila6** [comment]
I added a md file as a buffer. Hope it helps. Would love a pop up styled one, need to read docs.

tasks.json:
```json
[
  {
    "label": "Open Project Scratch",
    "command": "mkdir -p .zed && touch .zed/scratch.md && zed .zed/scratch.md",
    "hide": "on_success",
    "use_new_terminal": false,
    "reveal": "never",
  },
]
```

keymap.json:
```json
[
  {
    "context": "Editor && VimControl && !VimWaiting",
    "bindings": {
      "space .": ["task::Spawn", { "task_name": "Open Project Scratch" }],
    },
  },
]
```
